contracts/multisig-prover/src/contract/execute.rs

```diff
@@ -34,18 +34,22 @@ pub fn construct_proof(
         config.chain_name.clone(),
     )?;
 
-    let payload = match PAYLOAD
+    let payload = Payload::Messages(messages);
+    let payload_id = payload.id();
+
+    match PAYLOAD
         .may_load(deps.storage, &payload_id)
         .map_err(ContractError::from)?
     {
-        Some(payload) => payload,
+        Some(stored_payload) => {
+            if stored_payload != payload {
+                return Err(report!(ContractError::PayloadMismatch));
```

Comment:
might be useful to return the original payload here

Reply:
the stored payload I mean

Reply:
attached as a string, since implementing Display for Payload is trickier